extension/observer/dockerobserver/extension.go

```diff
@@ -36,17 +36,15 @@ type dockerObserver struct {
 	*observer.EndpointsWatcher
 	logger  *zap.Logger
 	config  *Config
-	cancel  func()
-	once    *sync.Once
-	ctx     context.Context
+	cancel  context.CancelFunc
+	wg      errgroup.Group
```

Comment:
Do we still need to store `cancel`? We can probably initialize `wg` using `WithContext` instead